demos/text_detection_demo/src/text_recognition.cpp

```diff
@@ -25,6 +26,25 @@ namespace  {
         }
         *prob = 1.0f / static_cast<float>(sum);
     }
+
+    void softmax(const std::vector<float>::const_iterator& begin, const std::vector<float>::const_iterator& end, std::vector<float> &prob) {
```

Comment:
Also one more minor change to improve style a bit:
```suggestion
    std::vector<float> softmax(const std::vector<float>::const_iterator& begin, const std::vector<float>::const_iterator& end) {
```
If the function returns probabilities instead of using a reference parameter, and returns as a value `std::vector<float>`, not as a reference `std::vector<float>&` (NB: t is impoirtant!), then the code will looks cleaner and slightly more optimal  -- it won't create a temporary object due to copy elision and return value optimization
(See details here: https://en.wikipedia.org/wiki/Copy_elision#Return_value_optimization)

In our case performance difference is not very important, so the change is more about the code style.